### DISCUSSION TITLE
Use thumbnails instead of originals

### DISCUSSION THREAD

**Discussion by dronnikovigor** [Radiokot/photoprism-android-client#116] [General]
Hi,
I was able to click app a bit and noticed some things that can be improved, although application is already cool.

So, i noticed, that when I open photo it opens originals - at least I heard that my disks started. 
As far as I know in web when I open photo - it takes max size thumbnail from photoprism cache on ssd and disks are off till I open a video. 
So, my thoughts was:
1. On photo click app opens max size thumbnail from photoprism cache OR takes from app cache original image
2. In background app loads original from server and upscale/override thumbnail. And being cached.

For now it takes about 1 sec to show photo, even for previously opened photo (maybe it's just spinner).

**Comment by Radiokot** [comment]
The app only load originals when you click "Download" or "Share".

Media viewer determines [photo preview size](https://docs.photoprism.app/user-guide/settings/advanced/#what-files-will-be-created-by-photoprism) based on your screen size and loads the corresponding image. If there is no cached preview of the requested size for the file in `storage/cache/thumbnails`, PhotoPrism generates it – this may be the cause of HDD activation. 

All the loaded images are cached in the app, but the cache has a size limit – the least recent used entries get removed. If you see that the app always loads images from the server, check if no redirects are happening for requests – if the first response is 301, 302, 307 or 308, the cache doesn't work.
